lib/view_component/base.rb

```diff
@@ -72,7 +72,13 @@ def render_in(view_context, &block)
       @current_template = self
 
       # Assign captured content passed to component as a block to @content
-      @content = view_context.capture(self, &block) if block_given?
+      if block_given?
+        if @content.nil?
```

Comment:
I think this may break existing cases where we render the same component instance twice.

e.g.

```ruby
my_component = MyComponent.new
render my_component { "foo" } # sets @content
render my_component { "bar" } # @content was set via the previous render call, but should now be overridden
```

What are your thoughts on setting an ivar like `@_content_set_by_with_content` or a similar approach?

Reply:
Good point. I've adjusted to use a similar method name and added a test to  ensure that multiple renders of the same instance will use the latest block provided.